test/threads.jl

```diff
@@ -341,3 +341,13 @@ end
 @testset "Base.Threads docstrings" begin
     @test isempty(Docs.undocumented_names(Threads))
 end
+
+@testset "jl_*affinity" begin
+    cpumasksize = @ccall uv_cpumask_size()::Cint
+    mask = zeros(Cchar, cpumasksize);
+    jl_getaffinity = (tid, mask, cpumasksize) -> ccall(:jl_getaffinity, Int32, (Int16, Ptr{Cchar}, Int32), tid, mask, cpumasksize)
+    jl_setaffinity = (tid, mask, cpumasksize) -> ccall(:jl_setaffinity, Int32, (Int16, Ptr{Cchar}, Int32), tid, mask, cpumasksize)
+    @test jl_getaffinity(1, mask, cpumasksize) == 0
+    fill!(mask, 1)
+    @test jl_setaffinity(1, mask, cpumasksize) == 0
```

Comment:
What happends with this test on MacOS?

> Thread affinity setting is not atomic on Windows. Unsupported on macOS.

Reply:
Oh, good catch. I'll test it locally and adjust it/turn it off on macOS if necessary.

Reply:
I've added a check that `cpumasksize` must be positive. Otherwise affinites are not supported on this platform (`UV_ENOTSUP`).